sktime/transformations/series/theta.py

```diff
@@ -32,28 +42,37 @@ def transform(self, Z, X=None):
 
         Returns
         -------
-        pd.DataFrame
-            Theta lines[1].
+        thetas: pd.DataFrame
+            Transformed series. DataFrame with Theta-lines[1]
+            where number of rows = len(Z), number of columns
+            = len(self.theta).
 
         References
         ----------
         [1] E.Spiliotis et al., "Generalizing the Theta method for
         automatic forecasting ", European Journal of Operational
         Research, vol. 284, pp. 550-558, 2020.
         """
+        self.check_is_fitted()
+        z = check_series(Z, enforce_univariate=True)
+
+        if len(self.theta) < 2:
```

Comment:
I suggest to write a helper function `_check_theta` for this so that users can either pass a single theta or a list or tuple. If users pass a single float, I suggest to still wrap it in a list internally, so that we don't have to change the for-loop below. Should we also enforce non-negative values?

Reply:
Thanks so much for the feedback, @mloning !
I thought that it doesn't make sense for ThetaForecaster to have less than two Theta-lines. Shall we wrap it in a list and add second element if a single theta is passed?
I don't know if anybody would like to use negative values, because they basically gives the same transformation as corresponding positive value, just reflected over linear regression. 
<img width="300" alt="Untitled" src="https://user-images.githubusercontent.com/73598322/122447050-9ddc2000-cfa3-11eb-8ca7-e1c59e2a228f.png">

Reply:
* Single theta. The thetaforecaster always has at least two lines, but maybe there are other situations where a user only wants to create a single theta line?
* Negative theta. I also think it is unlikely that negative theta is useful, so worth enforcing theta>=0.

Reply:
I wouldn't actually enforce theta >=0 (why? the line is perfectly defined), but I agree on one line

Reply:
the "recommendations" are encoded in sensible defaults, we shouldn't restrict the potential choices purely on a vague belief that it might not be useful (as opposed to, e.g., it being mathematically nonsensical).

Reply:
@fkiraly thank you for review! The thinking I had was this: the scenario when user knowingly passes negative theta in order to get the mirrored series seemed way less probable than one when someone passes negative values by mistake and gets mirrored forecasts that don't make sense. It's non mathematically nonsensical though, so I absolutely agree that the defaults should be enough in this case.

Reply:
well, in my opinion, it's better to let 9 users pass a negative value by mistake than nanny the 1 user who may want to do this, potentially for a very good reason (e.g., trying it out in a sensible tuning composite)

But I think this might be a matter of opinion and design principle - about how "opinionated" `sktime` should be in prescribing what it thinks are the best choices. I'm more "laissez faire" on this point...